core/src/main/java/hudson/FilePath.java

```diff
@@ -2627,7 +2673,8 @@ private Integer writeToTar(File baseDir, DirScanner scanner, OutputStream out) t
      * Reads from a tar stream and stores obtained files to the base dir.
      * Supports large files > 10 GB since 1.627 when this was migrated to use commons-compress.
      */
-    private void readFromTar(String name, File baseDir, InputStream in) throws IOException {
+    private List<String> readFromTar(String name, File baseDir, InputStream in) throws IOException {
```

Comment:
why dot pass the consume in here directly - so that you can get progress as it happens as opposed to after the fact (which is unhelpfull in the case something goes wrong)?